### ISSUE TITLE
TypeError: 'numpy.float64' object cannot be interpreted as an index

### ISSUE THREAD

**Issue by dmitrymurashenkov** [justinsalamon/melosynth#2]
I'm not familiar with python, but it seems that after python 1.11 float indexes cannot be used to access arrays or something. And this happens in script with following trace:

INFO:Loading data...
INFO:Generating wave...
Traceback (most recent call last):
  File "./melosynth.py", line 400, in <module>
    args.square, args.useneg)
  File "./melosynth.py", line 315, in melosynth
    freq_series = np.ones(nsamples) * f_prev
  File "/usr/lib/python2.7/dist-packages/numpy/core/numeric.py", line 192, in ones
    a = empty(shape, dtype, order)
TypeError: 'numpy.float64' object cannot be interpreted as an index

Can be fixed replacing line 308, but I'm not sure if this is the correct way to fix it:

-        nsamples = np.round((t - t_prev) * fs)
+        nsamples = int(np.round((t - t_prev) * fs))

**Comment by justinsalamon** [comment]
Thanks for reporting this! Please provide the exact command you used to call the script, and the versions of your operating system and numpy (I see you're on python 2.7). I'll have time to fix this in 2-3 weeks.

**Comment by dmitrymurashenkov** [comment]
Command:
./melosynth.py melody.csv

melody.csv content:
0.0,0
0.02,0
0.04,0

System: Linux Ubuntu 17.04
python package version: 2.7.13-2
python-numpy package version: 1:1.12.1-1ubuntu1

**Comment by beantowel** [comment]
to make the code compatible with python 3.6, i made following changes:
- line 135: 
```python
signal = b''.join((wave.struct.pack('h', int(item))) for item in x)
```
- line 138: 
```python
sampwidth = int(N / 8) # in bytes
```
- line 308: 
```python
nsamples = int(np.round((t - t_prev) * fs))
```

**Comment by justinsalamon** [comment]
Thanks @dmitrymurashenkov for reporting and thanks @beantowel for the suggested fix. I've just pushed the fix to version 0.1.1 of melosynth (now on master).
